src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs

```diff
@@ -101,14 +101,30 @@ public void TestChangedTextWithMultipleUnorderedChanges()
             var text = SourceText.From("Hello World");
             var changes = new[]
             {
-                new TextChange(new TextSpan(6, 7), "Universe"),
+                new TextChange(new TextSpan(6, 5), "Universe"),
                 new TextChange(new TextSpan(0, 5), "Halo")
             };
 
             var newText = text.WithChanges(changes);
             Assert.Equal("Halo Universe", newText.ToString());
         }
 
+        [Fact]
+        public void TestChangedTextWithMultipleUnorderedChangesAndOneIsOutOfBounds()
+        {
+            var text = SourceText.From("Hello World");
+
+            Assert.ThrowsAny<ArgumentException>(() =>
+            {
+                var changes = new[]
+                {
+                    new TextChange(new TextSpan(6, 7), "Universe"),
+                    new TextChange(new TextSpan(0, 5), "Halo")
+                };
+                var newText = text.WithChanges(changes);
+            });
```

Comment:
Consider limiting the section of code inside the Assert.Throws for clarity

```suggestion
            var changes = new[]
            {
                new TextChange(new TextSpan(6, 7), "Universe"),
                new TextChange(new TextSpan(0, 5), "Halo")
            };
            Assert.ThrowsAny<ArgumentException>(() =>
            {
                var newText = text.WithChanges(changes);
            });
```

Reply:
Sure :)